File: Niveau 1/52 - Type d'arbres/program.c
#include <stdio.h>

int main()
{
   int hauteur, folioles;
   scanf("%d%d", &hauteur, &folioles);
   
   if (hauteur <= 5 && folioles >= 8)
   {
      printf("Tinuviel");
   }
   if (hauteur >= 10 && folioles >= 10)
   {
      printf("Calaelen");
   }
   if (hauteur <= 8 && folioles <= 5)
   {
      printf("Falarion");
   }
   if (hauteur >= 12 && folioles <= 7)
   {
      printf("Dorthonion");
   }
 
   return 0;
}

File: Niveau 1/52 - Type d'arbres/program.py
hauteur = int(input())
nbFoliolles = int(input())
if hauteur <= 8:
   if nbFoliolles <= 5:
      print("Falarion")
   else:
      print("Tinuviel")
else:
   if nbFoliolles <= 7:
      print("Dorthonion")
   else:
      print("Calaelen")
